plugins/uefi-capsule/fu-uefi-tool.c

```diff
@@ -234,19 +236,23 @@ main(int argc, char *argv[])
 
 	/* override the default ESP path */
 	if (esp_path != NULL) {
-		esp = fu_volume_new_esp_for_path(esp_path, &error);
-		if (esp == NULL) {
+		g_autoptr(FuVolume) volume = NULL;
+		volume = fu_volume_new_esp_for_path(esp_path, &error);
+		if (volume == NULL) {
 			/* TRANSLATORS: ESP is EFI System Partition */
 			g_print("%s: %s\n", _("ESP specified was not valid"), error->message);
 			return EXIT_FAILURE;
 		}
-	} else {
-		esp = fu_volume_new_esp_default(&error);
-		if (esp == NULL) {
-			g_printerr("failed: %s\n", error->message);
-			return EXIT_FAILURE;
-		}
+		fu_context_add_esp_volume(ctx, volume);
+	}
+
+	/* get the default ESP only */
+	esp_volumes = fu_context_get_esp_volumes(ctx, &error);
+	if (esp_volumes == NULL) {
+		g_printerr("%s\n", error->message);
+		return EXIT_FAILURE;
 	}
+	esp = g_object_ref(g_ptr_array_index(esp_volumes, 0));
```

Comment:
same heuristic comment applies here.  Which means maybe we want the heuristic/message/whatever decision in a common symbol used by both places.

Reply:
At least for this tool; the other option is to prompt for ESP if multiple are found.

Reply:
I think in general we want to deal with multiple ESPs properly (i.e. your prompt idea) -- I don't think any heuristic is a good idea when we have an interactive user. Fixed.